crates/matrix-sdk-base/src/client.rs

```diff
@@ -1169,7 +1169,14 @@ impl BaseClient {
         // because we want to have the push rules in place before we process
         // rooms and their events, but we want to create the rooms before we
         // process the `m.direct` account data event.
-        self.handle_account_data(&response.account_data.events, &mut changes).await;
+        if !response.account_data.is_empty() {
```

Comment:
Still not correct, consider the following scenario:

We get some updated account data events, but the `m.direct` event was not updated. So `response.account_data` will not be empty, but it will also not contain the `m.direct` event. Thus rooms will remain without their direct target.

Reply:
I hope https://github.com/matrix-org/matrix-rust-sdk/pull/3990/commits/b53fa163922365d0ec021c4b2a767b7ff3d6e02b handles that in a better way 🤞 .